enhancements/network/egress-ip-multi-nic.md

```diff
@@ -0,0 +1,279 @@
+
+Another example, lets say we have the following 2 nodes with networks:
+```shell
+# worker 1
+3: ens4: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc fq_codel state UP group default qlen 1000
+link/ether 52:54:00:c7:bb:91 brd ff:ff:ff:ff:ff:ff
+altname enp0s4
+inet 192.168.123.181/16 brd 192.168.123.255 scope global dynamic noprefixroute ens4
+valid_lft 3552sec preferred_lft 3552sec
+inet6 fe80::a4ae:c1cb:2143:bb63/64 scope link noprefixroute
+valid_lft forever preferred_lft forever
+
+# worker 2
+3: ens4: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc fq_codel state UP group default qlen 1000
+link/ether 52:54:00:06:d6:58 brd ff:ff:ff:ff:ff:ff
+altname enp0s4
+inet 192.168.123.94/24 brd 192.168.123.255 scope global dynamic noprefixroute ens4
+valid_lft 3494sec preferred_lft 3494sec
+inet6 fe80::21df:c25a:dc63:f6eb/64 scope link noprefixroute
+valid_lft forever preferred_lft forever
+```
+
+Cluster manager would assign the `EgressIP` to worker 2.
+
+To allow packets to be steered into secondary networks, OVN reroute logical policies on `ovn_cluster_router` will redirect
+the packet out to the management port of the node which the egress IP is hosted.
+
+We will add rules in linux routing policy database control for each source Pod IP and lookup the correct routing table
+which will contain one route. This single route within a new managed routing table will direct the packet to the correct interface.
+
+We will also add a masquerade rule to iptables when a packet leaving a particular interface matches the source pod IP.
+
+
+See section `Design Details` for full implementation details.
+
+### User Stories
+As a user, I want the ability for cluster workload traffic to be routed to a network that is not managed by OVN and to have a configurable source IP.
+
+### Test Plan
+E2E tests verifying the correct source IP for deployment modes global IC and multi-zone IC:
+- Day 2 Networks CRUD including mask resizing
+- Configure new EgressIP both on ovn managed and unmanaged networks
+- Migrate existing EgressIP on ovn managed and unmanaged networks
+- Delete existing EgressIP on ovn managed and unmanaged networks
+- Add node
+- Delete node
+- Edit node
+- Add pod
+- Delete pod
+- Remove pod
+- Delete a valid rule manually 
+- Delete a valid IPtables rule manually
+- Delete a valid route manually
+- Delete a valid egress IP assigned to an interface manually
+- Valid egress service and egress IP functionality if both are active on the same k8 objs.
+Today, if egress service and egress IP both 'match' on the same pod, egress service is preferred.
+Continue this for this enhancement where egress service and egress IP overlap, prefer egress service.
+- Egress IP interface is now a slave to a bond which is now the IP of the EIP interface. Reconcile to move the EIP to the bond.
+- Egress IP interface is now a slave to a bond which is now a different IP of the original EIP interface. Reconcile to move the EIP
```

Comment:
let's replace `slave` with something more inclusive.

member of a bond ?... I honestly don't know what's used now in the kernel.

Reply:
changed it to child